Sources/Sentry/SentryNSFileManagerSwizzling.m

```diff
@@ -0,0 +1,84 @@
+#import "SentryNSFileManagerSwizzling.h"
+#import "SentryCrashDefaultMachineContextWrapper.h"
+#import "SentryCrashMachineContextWrapper.h"
+#import "SentryCrashStackEntryMapper.h"
+#import "SentryDependencyContainer.h"
+#import "SentryFileIOTracker.h"
+#import "SentryInAppLogic.h"
+#import "SentryNSProcessInfoWrapper.h"
+#import "SentryOptions+Private.h"
+#import "SentryStacktraceBuilder.h"
+#import "SentrySwizzle.h"
+#import "SentryThreadInspector.h"
+#import <SentryLog.h>
+#import <objc/runtime.h>
+
+@interface SentryNSFileManagerSwizzling ()
+
+@property (nonatomic, strong) SentryFileIOTracker *dataTracker;
+
+@end
```

Comment:
`m`: What drove you to create an extra class for `SentryNSFileManagerSwizzling`? Wouldn't it be better to have one class for swizzling file IO operations for tracing? Now, we have two different instances of the `dataTracker` and some duplicated logic.

Reply:
The main reason was future-proofing. NSFileManager offers many other methods which could be useful to auto-trace further down the road, i.e.:

- [contentsOfDirectoryAtPath:error:](https://developer.apple.com/documentation/foundation/nsfilemanager/1414584-contentsofdirectoryatpath) → `file.read`?
- [createDirectoryAtPath:withIntermediateDirectories:attributes:error:](https://developer.apple.com/documentation/foundation/nsfilemanager/1407884-createdirectoryatpath) → `file.write`?
- [removeItemAtURL:error:](https://developer.apple.com/documentation/foundation/nsfilemanager/1413590-removeitematurl) → `file.delete`
- [copyItemAtURL:toURL:error:](https://developer.apple.com/documentation/foundation/nsfilemanager/1412957-copyitematurl) → `file.rename`

That's why I believe its good to have separate swizzled classes into individual files.

But I do agree that the setup of multiple data trackers are not ideal. Two options:

1. Setup a shared data tracker in in `SentryFileIOTrackingIntegration.m` and pass it as an argument to `SentryNSDataSwizzling` and `SentryNSFileManagerSwizzling`.
2. Add the data tracker to `SentryDependencyContainer.m`

Reply:
I would prefer option 1.